site.py

```diff
@@ -0,0 +1,85 @@
+import os
+import subprocess
+import urllib
+import time
+import random
+from uuid import uuid4
+
+from flask import Flask, render_template, url_for, request, redirect, send_from_directory
+
+app = Flask(__name__)
+
+TEMP_DIR = 'tmp/'
+METADATA_FILE = 'metadata.csv'
+DATA_FILE = 'data.csv'
+
+# Retrieve the files uploaded by the user, and if they are valid,
+#  store them in a new generated subfolder of tmp/.
+@app.route('/', methods=['GET', 'POST'])
+def upload_file():
+    if request.method == 'GET':
+        return render_template('index.html')
+
+    if request.method == 'POST':
```

Comment:
`POST` is the only alternate method allowed in this route, so there is no need to nest it in a conditional.